src/replay/service.zig

```diff
@@ -91,6 +134,79 @@ fn advanceReplay(state: *ReplayState) !void {
     // TODO: maybe_start_leader
 }
 
+/// Identifies new slots in the ledger and starts tracking them in the slot
+/// tracker.
+///
+/// Analogous to
+/// [generate_new_bank_forks](https://github.com/anza-xyz/agave/blob/146ebd8be3857d530c0946003fcd58be220c3290/core/src/replay_stage.rs#L4149)
+fn trackNewSlots(
+    allocator: Allocator,
+    blockstore_db: *sig.ledger.BlockstoreDB,
+    slot_tracker: *SlotTracker,
+    epoch_tracker: *EpochTracker,
+    slot_leaders: SlotLeaders,
+    /// needed for update_fork_propagated_threshold_from_votes
+    _: *ProgressMap,
+) !void {
+    const root = slot_tracker.root.load(.monotonic);
+    var frozen_slots = try slot_tracker.frozenSlots(allocator);
+    defer frozen_slots.deinit(allocator);
+
+    var frozen_slots_since_root = try std.ArrayListUnmanaged(sig.core.Slot)
+        .initCapacity(allocator, frozen_slots.count());
+    defer frozen_slots_since_root.deinit(allocator);
+    for (frozen_slots.keys()) |slot| if (slot >= root) {
+        try frozen_slots_since_root.append(allocator, slot);
```

Comment:
Would it be better to use `appendAssumeCapacity` here given that the size would be less or greater to the init capacity of `frozen_slots.count()`?